src/common-helpers.ts

```diff
@@ -75,13 +75,29 @@ export const getMessageFromError = (error: Error): string => {
   return error.message;
 };
 
+/** Takes the original unvalidated examples from the properties of ZodObject schema shape */
+const pullExampleProps = <T extends z.SomeZodObject>(subject: T): T =>
```

Comment:
can I reduce the complexity of this function?
it has 2 reducers and 2 external calls

Reply:
I reused `copyMeta` because I didn't want to repeat `combinations`.
However, this one is simpler, and `combinations` does not need `deepMergeRight`, because we combine examples from individual props that do not intersect.